cfme/tests/v2v/test_v2v_osp.py

```diff
@@ -0,0 +1,105 @@
+"""Tests to validate openstack provider related usecases"""
+import fauxfactory
+import pytest
+from widgetastic.utils import partial_match
+
+from cfme import test_requirements
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.fixtures.provider import rhel7_minimal
+from cfme.fixtures.v2v_fixtures import get_migrated_vm
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE_PER_VERSION
+from cfme.utils.appliance.implementations.ui import navigate_to
+
+
+pytestmark = [
+    pytest.mark.provider(
+        classes=[OpenStackProvider],
+        selector=ONE_PER_VERSION,
+        required_flags=["v2v"],
+        scope="module",
+    ),
+    pytest.mark.provider(
+        classes=[VMwareProvider],
+        selector=ONE_PER_VERSION,
+        fixture_name="source_provider",
+        required_flags=["v2v"],
+        scope="module",
+    ),
+    test_requirements.v2v,
+    pytest.mark.usefixtures("v2v_provider_setup")
+]
+
+
+@pytest.mark.parametrize("project", ["default", "secondary"])
+@pytest.mark.parametrize("attribute", ["flavor", "security_group"])
+@pytest.mark.parametrize(
+    "mapping_data_vm_obj_single_datastore", [["nfs", "nfs", rhel7_minimal]], indirect=True)
+def test_v2v_custom_project_attribute(
+        request, appliance, provider, project, attribute, mapping_data_vm_obj_single_datastore):
+    """
+    Test V2V with custom attributes of openstack provider projects
+
+    Polarion:
+        assignee: ytale
+        caseimportance: medium
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.10
+        casecomponent: V2V
+        initialEstimate: 1h
+    """
+    infrastructure_mapping_collection = appliance.collections.v2v_infra_mappings
+    mapping_data = mapping_data_vm_obj_single_datastore.infra_mapping_data
+
+    if project == "secondary":
+        # Changing target cluster (openstack project) to other than default
+        component = mapping_data["clusters"][0].targets
+        component.targets.pop()
+        component.targets.append(partial_match(provider.data.clusters[1]))
+        map_cluster = component.targets[1]
+
+    mapping = infrastructure_mapping_collection.create(**mapping_data)
+    src_vm_obj = mapping_data_vm_obj_single_datastore.vm_list[0]
+
+    # Selecting default flavor and security group selection
+    map_flavor = provider.data.flavors[0]
+    map_security_group = provider.data.security_groups.admin[0]
```

Comment:
These are not used anywhere . These values are overridden by new values in line 75 and 77 . You can remove these

Reply:
No. Actually this are default values and we are only overridding those values when we have that specific `attribute` set by parameterization. Final permutation should be - 

[default cluster] + [flavor] + sec group
[default cluster] + flavor + [sec group]
[secondary cluster] + [flavor] + sec group
[secondary cluster] + flavor + [sec group]

Here values in square brakets are actual selection.